frontend/src/ProfilePage/Profile.jsx

```diff
@@ -107,17 +123,34 @@ function Profile() {
     }, [selectedOptions]);
     
     const getHistory = async (options) => {
-    // Fetch historical data for all selected options
-    const dataPromises = options.map(option =>
-        fetchHistoricalData(userId, option.value) // Pass the value of each option to the fetch function
-    );
-    const results = await Promise.all(dataPromises);
-    const combinedData = results.flat();
-    setChartData(combinedData); // Update the chart data with combined results from all selected options
-};
+        // Fetch historical data for all selected options
+        const dataPromises = options.map(option =>
+            fetchHistoricalData(userId, option.value) // Pass the value of each option to the fetch function
+        );
+        const results = await Promise.all(dataPromises);
+        const combinedData = results.flat();
+        setChartData(combinedData); // Update the chart data with combined results from all selected options
+    };
 
     const animatedComponents = makeAnimated(); 
 
+    /* when a new graph selection is made / deleted, set select options to match
+    the value of selectedOptions, tracked through the [options] parameter. 
+    The [selectedOptions] item in localStorage is also set to match this change
+    */
+    const handleSelectChange = (options) => {
+        setSelectedOptions(options);
+        localStorage.setItem('selectedOptions', JSON.stringify(options));
+    };
+
+    const saveSvgToLocalStorage = (svgElement) => {
```

Comment:
do we know how much storage can be stored in localStorage? How big are the svgs?

Reply:
I looked it up and it seems like in almost every browser localStorage can store up to 5MB. And to test the svg size,  I added a log statement to convert and log the size of the svg as it's being cached and it was always around 4800 bytes.